client/src/planwise/client/projects2/components/settings.cljs

```diff
@@ -382,7 +422,8 @@
           (map-indexed (fn [i iteration-step]
                          [:a {:key i
                               :class-name (join " " [(if (= (:step iteration-step) step) "active") (if (s/valid? (:spec iteration-step) project) "complete")])
-                              :href (routes/projects2-show-with-step {:id (:id project) :step (:step iteration-step)})}
+                              :href ((if read-only routes/projects2-settings-with-step routes/projects2-show-with-step)
+                                     {:id (:id project) :step (:step iteration-step)})}
                           (if (s/valid? (:spec iteration-step) project) [m/Icon "done"] [:i (inc i)])
                           [:div (:title iteration-step)]]) sections)]
          [m/GridCell {:span 6}
```

Comment:
There is a `(dispatch [:projects2/infer-step @current-project])` below here that is out of place and it is what was causing the navigation back to the scenarios list before this patch. You fixed this behavior by providing a default `review` step when showing the settings. But the step inference is a pure computation. So we can improve and simplify the code a bit further by removing the `dispatch` below, and inferring the current step if `nil` (using the `infer-step` function if editing or defaulting to `review` if showing) before looking for the `step-data`.

Reply:
@ggiraldez I removed the infer-step but I can't get rid of the dispatch. If the URL doesn't change then a step may become valid while typing and it will suddenly jump to the next step.